packages/amplify-graphql-transformer-core/src/utils/directive-wrapper.ts

```diff
@@ -34,15 +38,19 @@ export class DirectiveWrapper {
       arguments: this.arguments.map(arg => arg.serialize()),
     };
   };
-  public getArguments = <T>(defaultValue: Required<T>, featureFlags: FeatureFlagProvider): Required<T> => {
+  public getArguments = <T>(defaultValue: Required<T>, featureFlags?: FeatureFlagProvider, options?: GetArgumentsOptions): Required<T> => {
     const argValues = this.arguments.reduce(
       (acc: Record<string, any>, arg: ArgumentWrapper) => ({
         ...acc,
         [arg.name.value]: valueFromASTUntyped(arg.value),
       }),
       {},
     );
-    if (featureFlags.getBoolean('shouldDeepMergeDirectiveConfigDefaults', false)) {
+    const deepMergeFlag = 'shouldDeepMergeDirectiveConfigDefaults';
+    const useDeepMerge = featureFlags != null
```

Comment:
I would refactor this to just use the options instead of having 2 sources of truth here